keps/sig-network/2104-reworking-kube-proxy-architecture/README.md

```diff
@@ -0,0 +1,782 @@
+instance, introducing optimizations like EndpointSlice or new business semantics like Topology
+does not trigger a rebuild/release of any proxy implementation.
+
+The idea is to send the full state to the client, so implementations don't have to do
+diff-processing and maintain any internal state. This should provide simple implementations,
+reliable results and still be quite optimal, since many kernel network-level objects are
+updated via atomic replace APIs. It's also a protection from slow readers, since no stream has to
+be buffered.
+
+Since the node-local state computed by the new proxy will be simpler and node-specific, it will
+only change when the result for the current node is actually changed. Since there's less data in
+the local state, change frequency is reduced compared to cluster state. Testing on actual clusters
+showed a frequency reduction of change events by 2 orders of magnitude.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+TBD (Calico eBPF)
+
+#### Story 2
+
+TBD (node-local cluster DNS provider)
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- sending the full-state could be resource consuming on big clusters, but it should still be O(1) to
+  the actual kernel definitions (the complexity of what the node has to handle cannot be reduced
+  without losing functionality or correctness).
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+A [draft implementation] exists and some [performance testing] has been done.
+
+[draft implementation]: https://github.com/mcluseau/kube-proxy2/
+[performance testing]: https://github.com/mcluseau/kube-proxy2/blob/master/doc/proposal.md
+
+The watchable API will be a long polling, taking a "last known state info" and returning a stream of
+objects. Proposed definition:
```

Comment:
afaict these object do not exist in the current implementation

Reply:
ive updated the definition with existing `message` protos.... resolving